### PR TITLE
Ensure out-reference parameters of `CWallet::CreateTransaction` are initialized

### PR DESCRIPTION
**Pull request by daira**

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1348,8 +1348,8 @@ UniValue sendmany(const UniValue& params, bool fHelp)
 
     // Send
     CReserveKey keyChange(pwalletMain);
-    CAmount nFeeRequired = 0;
-    int nChangePosRet = -1;
+    CAmount nFeeRequired = -1; // never used
```
Comment (nuttycom) [inline src/wallet/rpcwallet.cpp:1351]: It makes me nervous to assign negative values to `CAmount` variables. Since this was already being set to zero, it wasn't uninitialized; why is this change needed?

**Review by nuttycom — CHANGES_REQUESTED** [review]
Blocking question about using negative `CAmount` values.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (daira) [reply src/wallet/rpcwallet.cpp:1351]: The particular change on this line isn't needed, because as the comment says, the value is never used. However, the change on 314 is needed and in that case, the initialization value can't be in-band: it has to be distinguishable from any valid `CAmount` value. I could have changed all of these uses to `std::optional<CAmount>` (which arguably is more semantically correct), but that would have been a more disruptive change than I wanted to make. I was being trying to be very systematic and ensure that exactly the same pattern is used for every call to `CreateTransaction`, which made it easier to have confidence that all cases had been caught.

Comment (nuttycom) [reply src/wallet/rpcwallet.cpp:1351]: Perhaps a safer way to handle these "never used" cases would be to declare them within a narrower scope, so that it's easier to see that they're not used outside that scope.

**Review by nuttycom — APPROVED** [review]
utACK with reservations.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (daira) [reply src/wallet/rpcwallet.cpp]: I followed this suggestion; please re-review.

**Review by nuttycom — APPROVED** [review]
utACK af00783f2bccc1cf61dc469ba82f9fc74e6c00d7

**Review by nuttycom — DISMISSED** [review]
re-utACK https://github.com/zcash/zcash/pull/6926/commits/35d55f1d40f5230b8e57a878ed5e1c22fa51e0cb

**Comment by nuttycom** [comment]
I have removed the wallet deprecation test changes from this PR, and put them over in https://github.com/zcash/zcash/pull/6934/commits/3b731828e1f7f44dbf13b31eaf77b94d6aadddfd because I saw a couple of what looked like indefinite hangs, probably caused by the `while True:` that was added.

**Review by nuttycom — APPROVED** [review]
utACK 4ca1c795dfaf7b85721ebb4597a7e4a00cb2a8d5
